connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -305,50 +309,63 @@ public void deleteConnector(String connName) throws IOException {
+            } catch(IOException e){
+                log.error("Could not parse connector list from response: {}",
+                        responseToString(response), e);
+                throw new ConnectException("Could not not parse connector list", e);
+            }
         }
+        throw new ConnectRestException(response.getStatus(),
+                "Could not read connector list. Error response: " + responseToString(response));
     }
 
     /**
      * Get the status for a connector running in this cluster.
      *
      * @param connectorName name of the connector
-     * @return an instance of {@link ConnectorStateInfo} populated with state informaton of the connector and it's tasks.
+     * @return an instance of {@link ConnectorStateInfo} populated with state information of the connector and its tasks.
      * @throws ConnectRestException if the HTTP request to the REST API failed with a valid status code.
      * @throws ConnectException for any other error.
      */
     public ConnectorStateInfo connectorStatus(String connectorName) {
         ObjectMapper mapper = new ObjectMapper();
+        String url = endpointForResource(String.format("connectors/%s/status", connectorName));
+        Response response = executeGet(url);
         try {
-            String url = endpointForResource(String.format("connectors/%s/status", connectorName));
-            return mapper.readerFor(ConnectorStateInfo.class).readValue(executeGet(url));
+            if (response.getStatus() < Response.Status.BAD_REQUEST.getStatusCode()) {
+                return mapper.readerFor(ConnectorStateInfo.class)
+                        .readValue(responseToString(response));
+            }
         } catch (IOException e) {
-            log.error("Could not read connector state", e);
-            throw new ConnectException("Could not read connector state", e);
+            log.error("Could not read connector state from response: {}",
+                    responseToString(response), e);
+            throw new ConnectException("Could not not parse connector state", e);
         }
+        throw new ConnectRestException(response.getStatus(),
+                "Could not read connector state. Error response: " + responseToString(response));
     }
 
-    public String adminEndpoint(String resource) throws IOException {
+    public String adminEndpoint(String resource) {
         String url = connectCluster.stream()
                 .map(WorkerHandle::adminUrl)
                 .filter(Objects::nonNull)
                 .findFirst()
-                .orElseThrow(() -> new IOException("Admin endpoint is disabled."))
+                .orElseThrow(() -> new ConnectException("Admin endpoint is disabled."))
                 .toString();
         return url + resource;
     }
 
-    public String endpointForResource(String resource) throws IOException {
+    public String endpointForResource(String resource) {
```

Comment:
Since the `IOException` was declared on the signature, it's clear that calling code has to handle this case. But because `ConnectException` is a runtime exception, we should probably add JavaDoc for this method to explain that `ConnectException` will be thrown if the endpoint is not known.

Reply:
Same as above